bash-preexec.sh

```diff
@@ -144,21 +147,38 @@ __bp_precmd_invoke_cmd() {
       return
     fi
     local __bp_inside_precmd=1
+    bash_preexec_invoke_precmd_functions "$__bp_last_ret_value" "$__bp_last_argument_prev_command"
+
+    __bp_set_ret_value "$__bp_last_ret_value" "$__bp_last_argument_prev_command"
+}
 
+# This function invokes every function defined in our function array
+# "precmd_function".  This function receives the arguments $1 and $2 for $? and
+# $_, respectively, that will be set for the precmd functions. This function
+# returns the last non-zero exit status of the hook functions. If there is no
+# error, this function returns 0.
+bash_preexec_invoke_precmd_functions() {
+    local lastexit=$1 lastarg=$2
     # Invoke every function defined in our function array.
     local precmd_function
+    local precmd_function_ret_value
+    local precmd_ret_value=0
```

Comment:
This is new functionality right? can we add a test for it on this new function. Not sure if it's covered by the existing coverage.

Reply:
Yes! I have added new tests for these new functions 918b451.